libraries/AP_Math/SplineCurve.cpp

```diff
@@ -0,0 +1,233 @@
+/*
+ This program is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <AP_HAL/AP_HAL.h>
+#include <AP_Math/AP_Math.h>
+#include <AP_InternalError/AP_InternalError.h>
+#include "SplineCurve.h"
+
+extern const AP_HAL::HAL &hal;
```

Comment:
we often leave the hal declaration in files because it makes for easier debugging.. I think it is one of the few places where we make an exception to the "all code must be used" rule.  I'm pretty sure we rejected a PR from someone who attempted to remove all the unused hal declarations.